src/Compiler/Driver/GraphChecking/Graph.fs

```diff
@@ -59,6 +75,48 @@ module internal Graph =
         |> readOnlyDict
         |> addIfMissing originalGraph.Keys
 
+    /// Returns leaves of the graph and the remaining graph without the leaves.
```

Comment:
Should be ok unless there's a project with _a lot_ of files. But it can probably be optimized together with `leafSequence`, it's already in the open tasks.